src/mono/wasm/host/WasmTestMessagesProcessor.cs

```diff
@@ -27,18 +29,26 @@ public void Invoke(string message)
             try
             {
                 logMessage = JsonSerializer.Deserialize<WasmLogMessage>(message);
-                line = logMessage?.payload ?? message.TrimEnd();
+                if (logMessage != null)
+                {
+                    line = logMessage.payload + " " + string.Join(" ", logMessage.arguments ?? Enumerable.Empty<object>());
```

Comment:
Why `Enumerable.Empty<object>` and not `Enumerable.Empty<string>`? Just curious, shouldn't matter though.

Reply:
I think you can also do `logMessage.arguments ?? []`

Reply:
`[]` will allocate every time, I think `Array.Empty<>` is what you want

Reply:
The arguments can contain any object. At least one call from worker init sends `{worker: 1}` as JS object. So I changed the arguments on apphost side to array of objects

Reply:
`Enumerable.Empty` has the same "caching frequently used values" as `Array.Empty` and enumerable is all I need. Is there any other reason to prefere array over enumerable?

Reply:
Enumerable.Empty is fine too then